### ISSUE TITLE
Crash when HEAD advanced during test

### ISSUE THREAD

**Issue by samueltardieu** [bors-ng/bors-ng#301]
I happened to change `master` while bors-ng was testing a branch. It looks like it didn't handle correctly the error when pushing afterwards.

Should it instead retry the PR (or add it to the next batch if other PRs are waiting, as was the case here)?

```
{{:badmatch, {:error, :push}},
 [{BorsNG.GitHub, :push!, 3,
   [file: 'lib/github.ex', line: 47]},
  {BorsNG.Worker.Batcher, :complete_batch, 3,
   [file: 'lib/batcher.ex', line: 373]},
  {BorsNG.Worker.Batcher, :maybe_complete_batch, 1,
   [file: 'lib/batcher.ex', line: 365]},
  {BorsNG.Worker.Batcher, :handle_cast, 2,
   [file: 'lib/batcher.ex', line: 71]},
  {:gen_server, :try_dispatch, 4,
   [file: 'gen_server.erl', line: 616]},
  {:gen_server, :handle_msg, 6,
   [file: 'gen_server.erl', line: 686]},
  {:proc_lib, :init_p_do_apply, 3,
   [file: 'proc_lib.erl', line: 247]}]}
```

**Comment by aaronjensen** [comment]
I just hit this though it was under different circumstances. The only thing that had pushed to master recently was bors.

**Comment by aaronjensen** [comment]
Ok, this is actually due to protected branches. In my case it was because I had codeclimate required, which does not run on bors staging commits. I've disabled protected branches for now (it's too bad we can't disable that big green merge button entirely...)

**Comment by notriddle** [comment]
If you turn protected branches back on, but only require `bors`, it'll disable the merge button while not interfering with bors.

**Comment by oreporan** [comment]
we're also experiencing this.
If master advances, bors will crash

**Comment by noizwaves** [comment]
I think this was fixed in #859

**Comment by notriddle** [comment]
Yes, it was. Thanks!
